include/grpcpp/impl/codegen/callback_common.h

```diff
@@ -203,7 +203,7 @@ class CallbackWithSuccessTag
     // callback, just as it silences a CQ tag in the async cases
     auto* ops = ops_;
     bool do_callback = ops_->FinalizeResult(&ignored, &ok);
-    GPR_CODEGEN_ASSERT(ignored == ops);
+    GPR_CODEGEN_DEBUG_ASSERT(ignored == ops);
```

Comment:
This change looks unrelated. What does it do?

Reply:
Per @vjpai :

"debug assert should be good
The story was just to make sure that we actually kept the same CQ tag between core and C++ layers in this particular use of FinalizeResult"

Reply:
Makes this path taken only on debug builds for even less performance cost.